### PR TITLE
Added validation function for add paybutton form

### PR DESCRIPTION
**Pull request by chedieck**
Input of paybutton addresses is now validated on the front end:
validating-input-2022-05-13_11 48 21

Legacy addresses are not being allowed.

### CODE REVIEW

**Comment by chedieck** [comment]
The width changes accordingly to the error message, it would be interest to fixate it the future

**Comment by alcipir** [comment]
IMO we should leave frontend for upcoming PRs since it will be restructured a bit. 
We could add validation on the backend as well though, just in case.

File: types/constants.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const supportedChains = [
```
Comment (Gattaca1) [inline types/constants.ts:1]: Supported chains should be requested from the server

**Comment by Gattaca1** [comment]
> IMO we should leave frontend for upcoming PRs since it will be restructured a bit. We could add validation on the backend as well though, just in case.

I'll second this. Lets add back end validation instead + error codes. Also lets do some sanitizing before the data gets put into the database (probably should have been part of yesterdays branch anyways)

File: types/constants.ts (hunk 1 above)
Comment (chedieck) [reply types/constants.ts:1]: You mean this should be a query that gets all chain slugs from the database? I thought about implementing that way, but that would be an extra query thatt would have to be done every time some address is input. Having a constant that corresponds to the supported chains in the database (this correspondence will be enforced by unittests) looked like a more efficient solution.

**Comment by chedieck** [comment]
I moved the validation logic to the backend.

**Comment by chedieck** [comment]
I've added unittests (and fixed the problems these unittests brought to the surface), now this PR depends on #72

**Comment by alcipir** [comment]
@Gattaca1

File: constants/index.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const supportedChains = [
+  'ecash',
+  'bitcoincash'
+] // WIP integration tests to check if this corresponds to the database
+export const supportedAddressesPattern = /(q|p)[a-z0-9]{41}/
```
Comment (Gattaca1) [inline constants/index.ts:5]: Lets get more specific with supported addresses. Heres the address format spec
[https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/cashaddr.cpp#L16](https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/cashaddr.cpp#L16)

relevant part: `const char *CHARSET = "qpzry9x8gf2tvdw0s3jn54khce6mua7l";`

File: constants/index.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const supportedChains = [
```
Comment (Gattaca1) [inline constants/index.ts:1]: Lets also add the testnets to the supported chains `bchtest`, `bchreg`

File: constants/index.ts (hunk 2 above)
Comment (Gattaca1) [reply constants/index.ts:5]: An equally valid charset is all uppercase. An address cannot mix upper and lower case though

`ecash:QPZRY9X8GF2TVDW0S3JN54KHCE6MUA7L` is valid
`ecash:QPZRY9X8GF2TVDW0S3jn54khce6mua7l` is not

**Comment by chedieck** [comment]
@Gattaca1 All remarks addressed.

File: utils/validators.test.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  )).toBe(true)
+});
+
+test('Accept example address one uppercase other lowercase', () => {
+  expect(v.validateAddresses(
+    ['ecash:QPZ274AAJ98XXNNKUS8HZV367ZA28J900C7TV5V8PC',
+     'bitcoincash:qz0dqjf6w6dp0lcs8cc68s720q9dv5zv8cs8fc0lt4',]
+  )).toBe(true)
+});
+test('Reject chain uppercase', () => {
```
Comment (Gattaca1) [inline utils/validators.test.ts:33]: Sorry man, I forgot to mention this from the spec. We should squash the chain to lowercase when we validate it, example here:
[https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/cashaddr.cpp#L264](https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/cashaddr.cpp#L264)

so `ECASH` becomes `ecash` making it valid

**Review by Gattaca1 — APPROVED** [review]
Looks good

**Pull request merged**